user-management.yaml

```diff
@@ -87,110 +88,34 @@ paths:
               schema:
                 $ref: '#/components/schemas/GenericError'
-  /users/{username}:
-    delete:
-      tags:
-      - "User Management"
-      summary: "Deletes a user"
-      description: "Deletes a user (when invoked by an administrator)"
-      security:
-        - uc4_auth: [] 
-      operationId: "deleteUser"
-      parameters:
-      - name: "username"
-        in: "path"
-        required: true
-        schema:
-          type: "string"
-      responses:
-        "200":
-          description: "OK"
-        "401":
-          description: |
-            Unauthorized  
-            types: authorization error
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/GenericError'
-        "403":
-          description: |
-            Forbidden  
-            types: not enough privileges
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/GenericError'
-        "404":
-          description: |
-            Not Found  
-            types: key not found
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/GenericError'
-  /role/{username}:  
-    get:
-      tags:
-      - "User Management"
-      summary: "Returns the role of a user"
-      description: "Returns the role of the user with the specified username"
-      security:
-        - uc4_auth: [] 
-      operationId: "getRole"
-      parameters:
-      - name: "username"
-        in: "path"
-        required: true
-        schema:
-          type: "string"
-      responses:
-        "200":
-          description: "OK"
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/Role'
-        "401":
-          description: |
-            Unauthorized  
-            types: authorization error
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/GenericError'
-        "404":
-          description: |
-            Not Found  
-            types: key not found
-          content:
-            application/json:
-              schema:
-                $ref: '#/components/schemas/GenericError'
-  
-  /students:
     post:
       tags:
-      - "Student Management"
-      summary: "Add a new student to the database"
-      description: "Adds a new student to the database after verification."
-      operationId: "addStudent"
+      - "User Management"
+      summary: "Add a new user to the database"
+      description: "Adds a new user to the database after verification."
+      operationId: "addUser"
       security:
         - uc4_auth: [] 
       requestBody:
-        description: "Student object that needs to be added to the database"
+        description: "User object that needs to be added to the database"
         required: true
         content:
```

Comment:
can we add an example value here? the UI does not generate one by default

same for all other endpoints, that use the oneOf notation

Reply:
Example values are now added in POST, PUT, and GET on /users endpoint